app/views/categories/_form.html.erb

```diff
@@ -0,0 +1,34 @@
+<div class="container-y container padding-20b">
+  <%= form_with(model: category, class: "form") do |form| %>
+    <% if category.errors.any? %>
+      <div style="color: red">
+        <h2><%= pluralize(category.errors.count, "error") %> prohibited this category from being saved:</h2>
+
+        <ul>
+          <% category.errors.each do |error| %>
+            <li><%= error.full_message %></li>
+          <% end %>
+        </ul>
+      </div>
+    <% end %>
+
+    <h1>New category</h1>
+
+    <%= form.hidden_field :user_id, value: current_user.id %>
+
+    <div>
+      <%= form.label :name, style: "display: block" %>
+      <%= form.text_field :name %>
+    </div>
+
+    <div>
+      <%= form.label :icon, style: "display: block" %>
+      <%= form.select :icon, [ ['🍺'], ['🍔'], ['💪'], ['🐺'], ['🐱'],
+                               ['✈'], ['💅'], ['🎮'], ['💊'], ['🩺'],
+                               ['💘'], ['💰'], ['💼'],[ '🎵'], ['🧦'] ]
+      %>
+    </div>
+
+    <%= form.submit class: "submit-btn" %>
+  <% end %>
+</div>
```

Comment:
While being able to create a new category and I think you missed a little requirement. 🤔 

 - "Add a new category" page

      - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
     - **The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success**.
     - When the user clicks on the "Back" button (<), the user navigates to the home page.
   
When I click on the save button, am not taken to the home page on success, but to a page, that shows that the category was successfully recorded( the category id route), however, when I then click the back button I am taken to the home page as required. 👍 

https://user-images.githubusercontent.com/66112760/205709192-ddd3d78e-1800-412e-8d63-f35c0273422e.mov


N.B: Please make this requested change also in creating new transactions(payment).

Reply:
Hi @bimbolabuari , thank you for your review. I did the changes as kindly required in both, the new category, and the new transaction. Now is working as expected. 😃